sui_core/src/authority.rs

```diff
@@ -517,12 +518,38 @@ impl AuthorityState {
             return Err(SuiError::TooManyItemsError(MAX_ITEMS_LIMIT));
         }
 
-        let (batches, transactions) = self
+        let (mut batches, transactions): (
+            Vec<SignedBatch>,
+            Vec<(TxSequenceNumber, TransactionDigest)>,
+        ) = self
             ._database
             .batches_and_transactions(request.start, request.end)?;
 
+        // Append the transaction info to individual transactions if requested
+        let mut transactions_with_info = Vec::new();
+        for (seq, digest) in transactions.iter() {
+            if request.include_tx_info {
+                let tx_info = self.make_transaction_info(digest).await?;
+                transactions_with_info.push((*seq, *digest, Some(Box::new(tx_info))));
+            } else {
+                transactions_with_info.push((*seq, *digest, None))
+            }
+        }
+
+        // Append the transaction info to batches if requested
+        if request.include_tx_info {
+            for mut signed_batch in &mut batches {
+                let mut transactions_with_info = Vec::new();
+                for (seq, digest, _) in &signed_batch.batch.transaction_batch.0 {
+                    let tx_info = self.make_transaction_info(digest).await?;
```

Comment:
Do the batches overlap with the set of transactions above?   I know that the subscriber API sends individual TX's then batches which include previous TXs.  I'm concerned that we are potentially requesting the transaction info's for duplicate TX's a second time here.

Reply:
Also doesn't the transaction batch need to be populated regardless of whether `include_tx_info` is set to True?